src/allotropy/parsers/thermo_fisher_nanodrop_eight/nanodrop_eight_parser.py

```diff
@@ -0,0 +1,272 @@
+    UltravioletAbsorbancePointDetectionMeasurementDocumentItems,
+)
+from allotropy.constants import ASM_CONVERTER_NAME, ASM_CONVERTER_VERSION
+from allotropy.named_file_contents import NamedFileContents
+from allotropy.parsers.thermo_fisher_nanodrop_eight.nanodrop_eight_reader import (
+    NanoDropEightReader,
+)
+from allotropy.parsers.vendor_parser import VendorParser
+
+T = TypeVar("T")
+
+CONCENTRATION_UNIT_TO_TQUANTITY = {
+    "ug/ul": TQuantityValueMicrogramPerMicroliter,
+    "ug/ml": TQuantityValueMicrogramPerMilliliter,
+    "mg/ml": TQuantityValueMilligramPerMilliliter,
+    "ng/ul": TQuantityValueNanogramPerMicroliter,
+    "ng/ml": TQuantityValueNanogramPerMilliliter,
+    "pg/ul": TQuantityValuePicogramPerMilliliter,
+}
+
+
+def _get_str(data_frame: pd.DataFrame, row: int, column: str) -> Optional[str]:
+    if column not in data_frame.columns:
+        return None
+    if pd.isna(data_frame.iloc[row][column]):
+        return None
+
+    return str(data_frame.iloc[row][column])
+
+
+def _get_float(data_frame: pd.DataFrame, row: int, column: str) -> JsonFloat:
+    try:
+        return float(data_frame.iloc[row][column])
+    except (ValueError, TypeError):
+        return InvalidJsonFloat.NaN
+
+
+def _get_concentration(conc: JsonFloat, unit: Optional[str]) -> Optional[T]:
+    if unit in CONCENTRATION_UNIT_TO_TQUANTITY and isinstance(conc, float):
+        cls = CONCENTRATION_UNIT_TO_TQUANTITY[unit]
+        return cls(value=conc)  # type: ignore[return-value]
+
+    return None
+
+
+class NanodropEightParser(VendorParser):
+    def to_allotrope(self, named_file_contents: NamedFileContents) -> Model:
+        contents, filename = named_file_contents
+        data = NanoDropEightReader.read(contents)
+        data = self._add_measurement_uuids(data)
+        return self._get_model(data, filename)
+
+    def _get_model(self, data: pd.DataFrame, filename: str) -> Model:
+        return Model(
+            field_asm_manifest="http://purl.allotrope.org/manifests/spectrophotometry/BENCHLING/2023/12/spectrophotometry.manifest",
+            spectrophotometry_aggregate_document=SpectrophotometryAggregateDocument(
+                spectrophotometry_document=self._get_spectrophotometry_document(data),
+                calculated_data_aggregate_document=CalculatedDataAggregateDocument(
+                    calculated_data_document=self._get_calculated_data_document(data),
+                ),
+                data_system_document=DataSystemDocument(
+                    file_name=filename,
+                    ASM_converter_name=ASM_CONVERTER_NAME,
+                    ASM_converter_version=ASM_CONVERTER_VERSION,
+                ),
+                device_system_document=DeviceSystemDocument(
+                    model_number="Nanodrop Eight",
+                    device_identifier="Nanodrop",
+                ),
+            ),
+        )
+
+    def _add_measurement_uuids(self, data: pd.DataFrame) -> pd.DataFrame:
```

Comment:
Since we're modifying `data` rather than creating a copy, I'd prefer not to have a return value.

Reply:
I was mainly worried that inPlace modification of dataframes seems to be generally frowned upon: https://github.com/pandas-dev/pandas/issues/16529

Reply:
ok, let's leave as is then. Thanks for future-proofing!